internal/providers/pluginfw/products/library/resource_library.go

```diff
@@ -35,28 +38,42 @@ func ResourceLibrary() resource.Resource {
 	return &LibraryResource{}
 }
 
-func readLibrary(ctx context.Context, w *databricks.WorkspaceClient, waitParams compute.Wait, libraryRep string, libraryExtended *LibraryExtended) diag.Diagnostics {
+var clusterDoesNotExistRegex = regexp.MustCompile(`Cluster [a-z1-9-]+ does not exist`)
+
+// readLibrary reads the status of the specified library on the specified cluster and returns the library metadata.
+// If library cannot be found, either because the cluster doesn't exist, the library is not installed, or some other error, the first return value will be nil.
+// The returned diagnostics will contain any errors or warnings that occurred during the operation, and the caller should check for errors before continuing.
+func readLibrary(ctx context.Context, w *databricks.WorkspaceClient, waitParams compute.Wait, libraryRep string) (*LibraryExtended, diag.Diagnostics) {
+	var d diag.Diagnostics
 	res, err := libraries.WaitForLibrariesInstalledSdk(ctx, w, waitParams, libraryDefaultInstallationTimeout)
+	var apierr *apierr.APIError
+	if errors.As(err, &apierr) && clusterDoesNotExistRegex.MatchString(apierr.Message) {
+		d.AddWarning("cluster not found", fmt.Sprintf("cluster %s not found", waitParams.ClusterID))
+		return nil, d
+	}
```

Comment:
If the cluster doesn't exist. we need to mark this resource as not existing to force recreation - in SDK, it was done by setting the ID to an empty string. How it could be done in plugin framework?

Reply:
Great point. That still happens, but it uses State's RemoveResource method: https://github.com/databricks/terraform-provider-databricks/pull/4280/files#diff-8f0d49da80872819daa7eb548715a81ee693d7b05bc657cd160005d4995c0eacR202